lib/c/string.zig

```diff
@@ -0,0 +1,45 @@
+const builtin = @import("builtin");
+const std = @import("std");
+const common = @import("common.zig");
+
+comptime {
+    @export(&strcmp, .{ .name = "strcmp", .linkage = common.linkage, .visibility = common.visibility });
+    @export(&strlen, .{ .name = "strlen", .linkage = common.linkage, .visibility = common.visibility });
+    @export(&strncmp, .{ .name = "strncmp", .linkage = common.linkage, .visibility = common.visibility });
+}
+
+fn strcmp(s1: [*:0]const c_char, s2: [*:0]const c_char) callconv(.c) c_int {
```

Comment:
You can make this u8 instead of c_char since it's the same ABI (what's being passed is a pointer), and avoid all the ptr casts in this file.

Reply:
Wouldn't that become an impediment to #20654 because we'd lose the information that the type in the signature is actually meant to be C `char`, which can be `u8` or `i8` depending on target?

Reply:
mm, good point. I did mention this though:

> Check for strict equality? Should there be a notion of compatibility? Should there be compatibility-checking settings? I'm thinking that there is indeed a notion of compatibility. For example, if the ABI defines a usize but you want to have an enum on the other side, that is well-defined behavior according to the ABI. However, one might want to tweak what kinds of compatibility are or are not counted as mismatch.

still I think you're right, the ability to generate viable .h files, or at least partial, would be nice.